### PR TITLE
[Feature] Support camera class for EG3D

### PR DESCRIPTION
**Pull request by LeoXing1996**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

EG3D needs a camera class to sample input conditions (camera-to-world matrix and intrinsic matrix).

## Modification

Support a few camera classes.

## TODO

- [x] Unit test for the camera.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmedit/models/editors/eg3d/camera.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+from mmengine.model import BaseModule
+
+from mmedit.registry import MODULES
+
+DeviceType = Optional[Union[str, int]]
+VectorType = Optional[Union[list, torch.Tensor]]
+
+
+class BaseCamera(BaseModule):
+    """Base camera class. Sample camera on sphere with specific distribution.
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:15]: sample camera position and return some parameters?

Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:15]: include "gaussian, uniform distribution"

File: mmedit/models/editors/eg3d/camera.py [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+import torch
+from mmengine.model import BaseModule
+
+from mmedit.registry import MODULES
+
+DeviceType = Optional[Union[str, int]]
+VectorType = Optional[Union[list, torch.Tensor]]
+
+
+class BaseCamera(BaseModule):
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:14]: How about removing BaseModule and related functions.

File: mmedit/models/editors/eg3d/camera.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+            Defaults to 'Uniform'.
+    """
+
+    def __init__(self,
+                 horizontal_mean: Optional[float] = None,
+                 vertical_mean: Optional[float] = None,
+                 horizontal_std: Optional[float] = 0,
+                 vertical_std: Optional[float] = 0,
+                 look_at: VectorType = [0, 0, 0],
+                 FOV: Optional[float] = None,
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:44]: fov

File: mmedit/models/editors/eg3d/camera.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+                         fov: Optional[float] = None,
+                         focal: Optional[float] = None,
+                         device: Optional[DeviceType] = None,
+                         batch_size: int = 1) -> torch.Tensor:
+        """Sample intrinsic matrix.
+
+        Args:
+            fov (Optional[float], optional): FOV (field of view) in degree. If
+                not passed, :attr:`self.fov` will be used. Defaults to None.
+            focal (Optional[float], optional): Focal in degree. If
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:100]:
```suggestion
            focal (Optional[float], optional): Focal in pixel. If
```

File: mmedit/models/editors/eg3d/camera.py [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+                '\'fov\', \'focal\', \'self.fov\' and \'self.focal\' should '
+                'not be None neither.')
+
+        if fov is not None:
+            intrinstic = self.FOV_to_intrinsic(fov, device)
+        else:
+            intrinstic = self.focal_to_instrinsic(focal, device)
+        return intrinstic[None, ...].repeat(batch_size, 1, 1)
+
+    def FOV_to_intrinsic(self,
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:130]: fov?

File: mmedit/models/editors/eg3d/camera.py [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+        focal = self.focal if focal is None else focal
+        assert focal is not None, (
+            '\'focal\' and \'self.focal\' should not be None at the '
+            'same time.')
+        device = self.device if device is None else device
+        intrinsics = [[focal, 0, 0.5], [0, focal, 0.5], [0, 0, 1]]
+        intrinsics = torch.tensor(intrinsics, device=device)
+        return intrinsics
+
+    def cpu(self) -> BaseModule:
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:180]: remove

File: mmedit/models/editors/eg3d/camera.py [hunk 7]
```diff
@@ -0,0 +1,445 @@
...
+
+        Returns:
+            torch.Tensor: Sampled theta.
+        """
+        h = self._sample_in_range(mean, std, batch_size)
+        return h
+
+    def sample_phi(self, mean: float, std: float,
+                   batch_size: int) -> torch.Tensor:
+        """Sampling the phi (pitch).
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:229]: Why sampling pitch is different from sampling yaw?

File: mmedit/models/editors/eg3d/camera.py [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+    translation_matrix = torch.eye(
+        4, device=origin.device).unsqueeze(0).repeat(forward_vector.shape[0],
+                                                     1, 1)
+    translation_matrix[:, :3, 3] = origin
+    cam2world = (translation_matrix @ rotation_matrix)[:, :, :]
+    assert (cam2world.shape[1:] == (4, 4))
+    return cam2world
+
+
+def normalize_vecs(vectors: torch.Tensor) -> torch.Tensor:
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:436]: seems the same with ppl.py/normalize

File: mmedit/models/editors/eg3d/camera.py [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+        radius (Optional[float]): Radius of the sphere. Defaults to None.
+    """
+
+    def __init__(self,
+                 horizontal_mean: Optional[float] = None,
+                 vertical_mean: Optional[float] = None,
+                 horizontal_std: Optional[float] = 0,
+                 vertical_std: Optional[float] = 0,
+                 look_at: List = [0, 0, 0],
+                 FOV: Optional[float] = None,
```
Comment (plyfager) [inline mmedit/models/editors/eg3d/camera.py:353]:
```suggestion
                 fov: Optional[float] = None,
```
